From d3a65ed0e636df41d912f4d95103bdaba68b416f Mon Sep 17 00:00:00 2001
From: Clemens Terasa <BLmadman@gmx.de>
Date: Wed, 29 Mar 2023 20:34:16 +0200
Subject: [PATCH] drm: img-rogue: Fix the Makefile

The makefile tries to determine its directory. However, this might be
faulty when built using Nix or other build environments.

Try to fix this.
---
 drivers/gpu/drm/img/img-rogue/Makefile                      | 6 ++----
 drivers/gpu/drm/img/img-rogue/config_kernel.mk              | 3 ---
 .../drm/img/img-rogue/services/server/env/linux/Kbuild.mk   | 6 +++++-
 3 files changed, 7 insertions(+), 8 deletions(-)

diff --git a/drivers/gpu/drm/img/img-rogue/Makefile b/drivers/gpu/drm/img/img-rogue/Makefile
index ee37ef8cadbf..109592dec867 100755
--- a/drivers/gpu/drm/img/img-rogue/Makefile
+++ b/drivers/gpu/drm/img/img-rogue/Makefile
@@ -54,9 +54,7 @@
 
 INTERNAL_KBUILD_OBJECTS=pvrsrvkm.o
 INTERNAL_EXTRA_KBUILD_OBJECTS=""
-TOP:=$(dir $(realpath $(lastword $(MAKEFILE_LIST))))
-TOP:=$(TOP:/=)
-BRIDGE_SOURCE_ROOT=$(TOP)/generated/rogue
+BRIDGE_SOURCE_ROOT=$(srctree)/$(src)/generated/rogue
 TARGET_PRIMARY_ARCH=target_riscv64
 PVR_ARCH=rogue
 PVR_ARCH_DEFS=rogue
@@ -96,7 +94,7 @@ bridge_base := $(BRIDGE_SOURCE_ROOT)
 # -I$(OUT)/include -I$(TOP)/kernel/drivers/staging/imgtec
 ccflags-y += -D__linux__ -include $(srctree)/$(src)/config_kernel.h \
  -include $(srctree)/drivers/gpu/drm/img/kernel_config_compatibility.h \
- -I$(OUT)/include \
+ -I$(srctree)/$(src)/include \
  -I$(srctree)/drivers/gpu/drm/img \
  -I$(srctree)/$(src)/hwdefs/rogue \
  -I$(srctree)/$(src)/hwdefs/rogue/km
diff --git a/drivers/gpu/drm/img/img-rogue/config_kernel.mk b/drivers/gpu/drm/img/img-rogue/config_kernel.mk
index 503847443a33..fbba7252cccb 100644
--- a/drivers/gpu/drm/img/img-rogue/config_kernel.mk
+++ b/drivers/gpu/drm/img/img-rogue/config_kernel.mk
@@ -11,9 +11,6 @@ override PVR_ARCH := rogue
 override METAG_VERSION_NEEDED := 2.8.1.0.3
 override MIPS_VERSION_NEEDED := 2014.07-1
 override RISCV_VERSION_NEEDED := 1.0.1
-override KERNELDIR := /home/lisl/freelight-u-sdk/soft_3rdpart/IMG_GPU/linux/../../../work/linux
-override KERNEL_ID := 5.15.0-00006-g35a89f4886a9-dirty
-override PVRSRV_MODULE_BASEDIR := /lib/modules/5.15.0-00006-g35a89f4886a9-dirty/extra/
 override KERNEL_COMPONENTS := srvkm drm_starfive
 override KERNEL_CROSS_COMPILE := riscv64-linux-
 override WINDOW_SYSTEM := nulldrmws
diff --git a/drivers/gpu/drm/img/img-rogue/services/server/env/linux/Kbuild.mk b/drivers/gpu/drm/img/img-rogue/services/server/env/linux/Kbuild.mk
index 4ba7716ab8bf..c830a14146f7 100644
--- a/drivers/gpu/drm/img/img-rogue/services/server/env/linux/Kbuild.mk
+++ b/drivers/gpu/drm/img/img-rogue/services/server/env/linux/Kbuild.mk
@@ -39,6 +39,9 @@
 # CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 ### ###########################################################################
 
+TOP := ../drivers/gpu/drm/img/img-rogue
+#TOP := /build/source/drivers/gpu/drm/img/img-rogue
+
 # Window system
 ccflags-y += -DWINDOW_SYSTEM=\"$(WINDOW_SYSTEM)\"
 
@@ -51,6 +54,7 @@ ccflags-y += \
 ccflags-y += -I$(TOP)/services/shared/devices/$(PVR_ARCH_DEFS)
 
 # Errata files
+HWDEFS_DIR := $(TOP)/hwdefs/rogue
 ccflags-y += -I$(HWDEFS_DIR) -I$(HWDEFS_DIR)/$(RGX_BNC)
 
 # Linux-specific headers
@@ -621,4 +625,4 @@ ccflags-y += -Wno-ignored-qualifiers
 # Treat #warning as a warning
 ccflags-y += -Wno-error=cpp
 
-include $(SYSTEM_DIR)/Kbuild.mk
+#include $(SYSTEM_DIR)/Kbuild.mk
-- 
2.38.4

